source/src/cfg/cs_Boards.c

```diff
@@ -213,3 +223,77 @@ cs_ret_code_t configure_board(boards_config_t* config) {
 
 	return ERR_SUCCESS;
 }
+
+cs_ret_code_t configure_board_from_uicr(const cs_uicr_data_t* uicrData, boards_config_t* config) {
+	init(config);
+	config->hardwareBoard = uicrData->board;
+
+	switch (uicrData->productRegionFamily.fields.productType) {
+		case PRODUCT_DEV_BOARD:
+			asPca10040(config);
+			return ERR_SUCCESS;
+		case PRODUCT_CROWNSTONE_PLUG_ZERO: {
```

Comment:
Do you really want to run the risk not to recognize the products even though you have already recognized them as a PRODUCT_CROWNSTONE_PLUG_ZERO? What's the consequence for the user if they happen to have plugs with e.g. a minor version 2?